From: Stefano Zacchiroli <zack@debian.org>
Date: Fri, 25 Aug 2006 13:33:04 +0200
Subject: Add recognition of more LaTeX commands for tex filetype detection

Since filetype detection of TeX files defaults to plaintex, we've added
detection of some additional LaTeX commands to help sway the detection
to LaTeX.

Closes: #384479
Signed-off-by: James McCoy <jamessan@debian.org>
---
 runtime/autoload/dist/ft.vim | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/autoload/dist/ft.vim b/runtime/autoload/dist/ft.vim
index de45faa..6a4294c 100644
--- a/runtime/autoload/dist/ft.vim
+++ b/runtime/autoload/dist/ft.vim
@@ -665,7 +665,7 @@ func dist#ft#FTtex()
     call cursor(1,1)
     let firstNC = search('^\s*[^[:space:]%]', 'c', 1000)
     if firstNC " Check the next thousand lines for a LaTeX or ConTeXt keyword.
-      let lpat = 'documentclass\>\|usepackage\>\|begin{\|newcommand\>\|renewcommand\>'
+      let lpat = 'documentclass\>\|usepackage\>\|begin{\|newcommand\>\|renewcommand\>\|part\>\|chapter\>\|section\>\|subsection\>\|subsubsection\>\|paragraph\>\|subparagraph\>\|subsubparagraph'
       let cpat = 'start\a\+\|setup\a\+\|usemodule\|enablemode\|enableregime\|setvariables\|useencoding\|usesymbols\|stelle\a\+\|verwende\a\+\|stel\a\+\|gebruik\a\+\|usa\a\+\|imposta\a\+\|regle\a\+\|utilisemodule\>'
       let kwline = search('^\s*\\\%(' . lpat . '\)\|^\s*\\\(' . cpat . '\)',
 			      \ 'cnp', firstNC + 1000)
